ballerina/tests/jms_queue_tests.bal

```diff
@@ -0,0 +1,209 @@
+    'type: QUEUE,
+    name: "test-queue-1"
+});
+final MessageConsumer queue1Consumer = check createConsumer(AUTO_ACK_SESSION, destination = {
+    'type: QUEUE,
+    name: "test-queue-1"
+});
+
+@test:Config {
+    groups: ["queue"]
+}
+isolated function testQueueWithTextMessage() returns error? {
+    string content = "This is a sample message";
+    TextMessage message = {
+        content: content
+    };
+    check queue1Producer->send(message);
+    Message? response = check queue1Consumer->receive(5000);
+    test:assertTrue(response is TextMessage, "Invalid message type received");
+    if response is TextMessage {
+        test:assertEquals(response.content, content, "Invalid payload");
+    }
+}
+
+@test:Config {
+    groups: ["queue"]
+}
+isolated function testQueueWithMapMessage() returns error? {
+    map<anydata> content = {
+        "user": "John Doe",
+        "message": "This is a sample message"
+    };
+    MapMessage message = {
+        content: content
+    };
+    check queue1Producer->send(message);
+    Message? response = check queue1Consumer->receive(5000);
+    test:assertTrue(response is MapMessage, "Invalid message type received");
+    if response is MapMessage {
+        test:assertEquals(response.content, content, "Invalid payload");
+    }
+}
+
+@test:Config {
+    groups: ["queue"]
+}
+isolated function testQueueWithBytesMessage() returns error? {
+    byte[] content = "This is a sample message".toBytes();
+    BytesMessage message = {
+        content: content
+    };
+    check queue1Producer->send(message);
+    Message? response = check queue1Consumer->receive(5000);
+    test:assertTrue(response is BytesMessage, "Invalid message type received");
+    if response is BytesMessage {
+        test:assertEquals(response.content, content, "Invalid payload");
+    }
+}
+
+@test:Config {
+    groups: ["queue"]
+}
+isolated function testTempQueue() returns error? {
+    MessageProducer tempQueueProducer = check createProducer(AUTO_ACK_SESSION, {
+        'type: TEMPORARY_QUEUE
+    });
+    string content = "This is a sample message";
+    TextMessage message = {
+        content: content
+    };
+    check tempQueueProducer->send(message);
+    check tempQueueProducer->close();
+}
+
+@test:Config {
+    groups: ["queue"]
+}
+isolated function testQueueProducerSendToError() returns error? {
+    TextMessage message = {
+        content: "This is a sample message"
+    };
+    Error? result = queue1Producer->sendTo({
+        'type: QUEUE,
+        name: "unsupported-queue"
+    }, message);
+    test:assertTrue(result is Error, 
+        "Allowing to send messages to other destination rather than the configured destination");
+}
+
+final MessageProducer queueProducerWithoutDestination = check createProducerWithoutDestination(AUTO_ACK_SESSION);
+final MessageConsumer queue2Consumer = check createConsumer(AUTO_ACK_SESSION, destination = {
+    'type: QUEUE,
+    name: "test-queue-2"
```

Comment:
Can define a constant for "test-queue-2" since it's used in multiple places

Reply:
Since this value is local for one file and not going to change over time I don't see any additional advantage for doing that.